### PR TITLE
Workflow: Canary Test

### PR DESCRIPTION
**Pull request by aparna-ravindra**
### What are you trying to accomplish?

Daily run for E2E run of the CLI extension

### What approach did you choose and why?

For each runner os, the workflow
- creates a cache 
- ensures that the list command retrieves the cache
- ensures that the delete command can delete the cache

### Anything you want to highlight for special attention from reviewers?
N/A

### CODE REVIEW

**Review by t-dedah — COMMENTED** [review]
Shouldnt this file be part of [actions/canary](https://github.com/actions/canary) repo? Did we add this here in order to use whatever secrets we need?

File: .github/workflows/canary.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Canary Test
+
+on:
+  release:
+    types: [released]
+  schedule:
+  - cron: "30 18 * * *"
```
Comment (t-dedah) [inline .github/workflows/canary.yaml:7]: Should we change the schedule to be every four hours ?
`0 */4 * * *`

File: .github/workflows/canary.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          echo $cache_found
+          [[ -z "$cache_found" ]] && exit 1 || echo "Cache List Successful"
+      - name: Delete Command
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.REPO_WRITE_TOKEN }}
+        run: |
+          cache_delete=$(gh actions-cache delete $CacheKey --confirm | grep "Deleted 1 cache entry with key")
+          echo $cache_delete
+          [[ -z "$cache_delete" ]] && exit 1 || echo "Cache Delete Successful"
```
Comment (t-dedah) [inline .github/workflows/canary.yaml:58]: Should we add slack notification if the workflow fails?
How will we come to know about failure in the current workflow?

Comment (aparna-ravindra) [reply .github/workflows/canary.yaml:58]: done

**Review by t-dedah — APPROVED** [review]

**Pull request merged**